README.md

```diff
@@ -59,15 +54,50 @@ implement that:
 
     response = http.request_get(uri.path, Authorization: access_token.authorization_header)
 
+## Service Provider Use Case
+
+    # Acquire Authorization header value from HTTP server's request
+    authz_header = request['Authorization']
+
+    # Parse the header value
+    access_token = AccessToken.parse_authorization_header(authz_header)
+
+    # Authenticate the Access Token
+    # Note: An AccessTokenAgent, configured with a System Account that has been granted privileges
+    # to Acquire Tokens and Process Tokens.
+    begin
+      results = access_token.authenticate(agent)
+    rescue OAuthError => e
+      # respond with a 401
```

Comment:
In some cases the service provider should return a 400 (for example when the signature method is unsupported) [1]. Should there be something that helps service providers to determine the proper status code from the error?

[1] https://oauth.net/core/1.0/#rfc.section.10

Reply:
OAuthError has an oauth_problem attribute, which should allow for such a mapping.

We could add some methods to the Protocol class for taking an oauth_problem and converting that into a HTTP status code and WWW-Authenticate header value.

Reply:
I think that would be helpful. I know our current internal oauth1a provider gem does not set the correct status codes for all `oauth_problems`. Providing the mapping as part of this gem will make it a lot easier for providers to implement the correct behavior.

Reply:
I pushed some more changes to this branch related to converting errors/problems to HTTP status codes and WWW-Authenticate headers. Checkout the OAuthError#to_http_status and OAuthError#to_http_www_authenticate_header methods.